src/core/geometry/qgsgeometryutils_base.h

```diff
@@ -464,4 +464,91 @@ class CORE_EXPORT QgsGeometryUtilsBase
      * \since QGIS 3.34
      */
     static double azimuth( double x1, double y1, double x2, double y2 ) SIP_HOLDGIL;
+
+#ifndef SIP_RUN
+
+    /**
+     * Performs fuzzy comparison between pairs of values within a specified epsilon.
+     *
+     * This function compares a variable number of pairs of values to check if their differences
+     * fall within a specified epsilon range using qgsNumberNear. It returns true if all the differences
+     * are within the given epsilon range; otherwise, it returns false.
+     *
+     * \tparam T Floating-point type (double or float) for the values to be compared.
+     * \tparam Args Type of arguments for the values to be compared.
+     * \param epsilon The range within which the differences are checked.
+     * \param args Variadic list of values to be compared in pairs.
+     *             The number of arguments must be greater than 0 and even.
+     *             It must follow the pattern: x1, y1, x2, y2, or x1, y1, z1, x2, y2, z2, ...
+     * \return true if all the differences between pairs of values are within epsilon, false otherwise.
+     * \throws std::invalid_argument If the number of arguments is not greater than 0 or not even.
+     *
+     * \see fuzzyDistanceEqual
+     *
+     * \since QGIS 3.36
+     */
+    template<typename T, typename... Args>
+    static bool fuzzyEqual( T epsilon, const Args &... args )
+    {
+      constexpr size_t numArgs = sizeof...( args );
+      if ( numArgs % 2 != 0 || numArgs == 0 )
+      {
+        throw std::invalid_argument( "The number of arguments must be greater than 0 and even" );
```

Comment:
@nyalldawson
It's true that initially, I thought this could be used elsewhere, maybe even in Python; however, that's not the case in the end. So, might as well use 'static_assert.' As for 'noexcept,' it's just a keyword to add; in my opinion, it's not super useful, but it's better to include it when possible, as is the case here.